src/commands/StartingResourcesLogStep.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type ApplicationInsightsComponent } from "@azure/arm-appinsights";
+import { type AppServicePlan, } from "@azure/arm-appservice";
+import { type ResourceGroup } from "@azure/arm-resources";
+import { type ILocationWizardContext } from "@microsoft/vscode-azext-azureutils";
+import { ActivityChildItem, ActivityChildType, activityInfoContext, activityInfoIcon, AzureWizardPromptStep, createContextValue, type ExecuteActivityContext, type IActionContext } from "@microsoft/vscode-azext-utils";
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { prependOrInsertAfterLastInfoChild } from "../utils/activityUtils";
+
+type StartingResourcesLogContext = IActionContext & Partial<ExecuteActivityContext> & ILocationWizardContext & {
+    resourceGroup?: ResourceGroup,
+    plan?: AppServicePlan,
+    appInsightsComponent?: ApplicationInsightsComponent,
+};
+
+const startingResourcesContext: string = 'startingResourcesLogStepItem';
+
+export class StartingResourcesLogStep<T extends StartingResourcesLogContext> extends AzureWizardPromptStep<T> {
```

Comment:
This seems like something that belongs in the azure package. I made a similar comment to Matt in his Storage PR as well.

It almost feels like rather than being a step that's injected, it's something that can happen in the wizard itself-- maybe we have a list of keys or something that it checks for before starting to execute steps.

Reply:
Yah i just followed the conventions set from the Storage PR. I agree that it makes more sense for it to happens in the wizard itself. Thoughts @MicroFish91?

Reply:
Yeah that could work.  What if we ship this way for now and see what each client extension needs and then investigate replacing with a more generic shared solution after build?